src/main/java/com/work/management/service/team/TeamService.java

```diff
@@ -0,0 +1,5 @@
+package com.work.management.service.team;
+
+public interface TeamService<TeamDto, Team> {
+    void save(TeamDto document);
```

Comment:
this could just be public interface TeamService{
TeamDto save (TeamDto teamDto);
}

Try not making your service depends on Database entity class... if you decide to rename you database entity then you should be changing as little code as possible...